### PR TITLE
[CARBONDATA-1327] Add carbon sort column examples

### PR DESCRIPTION
**Pull request by mayunSaicmotor**
Add carbon sort column examples and update the introduction document

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3184/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/589/

File: docs/ddl-operation-on-carbondata.md [hunk 1]
```diff
@@ -101,6 +101,14 @@ The following DDL operations are supported in CarbonData :
 
    - All dimensions except complex datatype columns are part of multi dimensional key(MDK). This behavior can be overridden by using TBLPROPERTIES. If the user wants to keep any column (except columns of complex datatype) in multi dimensional key then he can keep the columns either in DICTIONARY_EXCLUDE or DICTIONARY_INCLUDE.
 
+   - **Sort Columns Configuration**
+
+      It is used to specify the  multi dimensional key(MDK) columns. By default MDK is composed of all dimension columns except complex datatype column. 
+
```
Comment (chenliang613) [inline docs/ddl-operation-on-carbondata.md]: here, need give the description for "SORT_COLUMN" property:
"SORT_COLUMN"  property is for users to specify which columns belong to the MDK index.
If user don't specify "SORT_COLUMN" property, by default MDK index be built by using all dimension columns except complex datatype column.

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonSortColumnsExample.scala [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    val rootPath = new File(this.getClass.getResource("/").getPath
+                            + "../../../..").getCanonicalPath
+    val storeLocation = s"$rootPath/examples/spark2/target/store"
+    val warehouse = s"$rootPath/examples/spark2/target/warehouse"
+    val metastoredb = s"$rootPath/examples/spark2/target"
+
+    CarbonProperties.getInstance()
+      .addProperty(CarbonCommonConstants.CARBON_TIMESTAMP_FORMAT, "yyyy/MM/dd HH:mm:ss")
+      .addProperty(CarbonCommonConstants.CARBON_DATE_FORMAT, "yyyy/MM/dd")
+      .addProperty(CarbonCommonConstants.ENABLE_UNSAFE_COLUMN_PAGE_LOADING, "true")
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonSortColumnsExample.scala]: can you explain ,why add : addProperty(CarbonCommonConstants.ENABLE_UNSAFE_COLUMN_PAGE_LOADING, "true")

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonSortColumnsExample.scala [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+    val path = s"$rootPath/examples/spark2/src/main/resources/data.csv"
+
+    // scalastyle:off
+    spark.sql(
+      s"""
+         | LOAD DATA LOCAL INPATH '$path'
+         | INTO TABLE no_sort_columns_table
+         | OPTIONS('FILEHEADER'='shortField,intField,bigintField,doubleField,stringField,timestampField,decimalField,dateField,charField,floatField,complexData',
+         | 'COMPLEX_DELIMITER_LEVEL_1'='#')
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonSortColumnsExample.scala]: Currently, sort_column don't support "float,double,decimal", please add the comment in this example, but can support other numeric type(like : int,long)

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/600/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3195/

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonSortColumnsExample.scala (hunk 3 above)
Comment (mayunSaicmotor) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonSortColumnsExample.scala]: added comments in line 74 and line 75

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3196/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/601/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/602/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3197/

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonSortColumnsExample.scala [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+         | floatField FLOAT,
+         | complexData ARRAY<STRING>
+         | )
+         | STORED BY 'carbondata'
+         | TBLPROPERTIES('SORT_COLUMNS'='')
+       """.stripMargin)
+
+    // Create table with sort columns
+    // Currently sort_column don't support "FLOAD, DOUBLE, DECIMAL"
+    // but can support other numeric type(like: INT, LONG)
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/CarbonSortColumnsExample.scala]: How about changing the below comments 
from 
// Currently sort_column don't support "FLOAD, DOUBLE, DECIMAL"
 +    // but can support other numeric type(like: INT, LONG)
to 
// you can specify any columns to sort columns for building MDX index, remark: currently sort columns don't support "FLOAT, DOUBLE, DECIMAL"

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/662/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3257/

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**